test/transcoding/atomics.spt

```diff
@@ -40,7 +40,7 @@
 7 AtomicXor 3 24 7 13 14 13 
 7 AtomicAnd 3 25 7 13 14 13 
 9 AtomicCompareExchange 3 26 7 13 14 14 13 17 
-9 AtomicCompareExchangeWeak 3 27 7 13 14 14 13 17 
+9 AtomicCompareExchange 3 27 7 13 14 14 13 17 
```

Comment:
Why do we need this change? While stopping generation of `OpAtomicCompareExchangeWeak` during LLVM IR to SPIR-V translation, we shouldn't break backward compatibility for previously generated SPIR-V modules.

Reply:
I agree that we need to emit OpAtomicCompareExchangeWeak, but selectively for SPIR-V 1.3 or earlier. 
Thanks

Reply:
Well, we may stop emitting the instruction, it's not a problem. The problem is, that once emitted - it should be supported by the translator during SPIR-V consumption. Just imagine being a user, who compiled once a valid OpenCL application in JIT mode one day with drivers update find out, that the application crashes during execution (I believe, that is what happening with the test now).
So, my suggestion is:
for LLVM IR to SPIR-V translation have
```
compare_exchange_strong_explicit -> AtomicCompareExchange
compare_exchange_weak_explicit -> AtomicCompareExchange
```

For reverse translation:
```
AtomicCompareExchange ->compare_exchange_strong_explicit
AtomicCompareExchangeWeak -> compare_exchange_weak_explicit
```

Reply:
I've read this #1230. The backward compatibility argument makes sense & broadens test coverage - so that's fine with me. I can update.

Reply:
_AtomicCompareExchangeWeak_ must _not_ be translated to _compare_exchange_weak_explicit_.

It can _only_ be safely translated to _compare_exchange_strong_explicit._

Reply:
@StuartDBrady yes, it was my misinterpretation. Current version of the patch doesn't follow it, see test/transcoding/atomics.spt test remained to be unchanged .

Reply:
@StuartDBrady friendly ping. The patch doesn't add this mapping to compare_exchange_weak_explicit

Reply:
@StuartDBrady I was talking about this test: https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/main/test/transcoding/atomics.spt
As you may see it has both `AtomicCompareExchange` and `AtomicCompareExchangeWeak` in pre-generated SPIR-V and then on line [92 and 93 ](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/main/test/transcoding/atomics.spt#L92) we have a check for "strong" OpenCL call. So given that this patch doesn't modify this test, it seems like the behavior is correct.